app/helpers/contents_list_helper.rb

```diff
@@ -0,0 +1,14 @@
+module ContentsListHelper
+  def insert_spans(text)
+    no_tags = strip_tags(text) #just the text of the link
+    number = /^[0-9]*[.]/.match(no_tags)
+
+    if number
+      link_text = no_tags.sub number.to_s, '' #remove the number from the text
+      text.sub! no_tags, '<span class="contents-number">' + number.to_s + '</span><span class="contents-text">' + link_text.strip + '</span>'
+      text = text.squish.html_safe
```

Comment:
It wouldn't, unfortunately. The reason is that the value of text is a link tag, so we're replacing the text within that tag with the text containing the spans. It would need to be something more like

> <a href="#"><span class= ... (your suggestion) ...</span></a>

...except we don't know what the link will be.